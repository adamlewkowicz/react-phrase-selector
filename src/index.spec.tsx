import React from 'react';
import { PhraseSelector } from '.';
import { render } from '@testing-library/react';

jest.mock('./style.css', () => ({}));

describe('<PhraseSelector />', () => {

  test('selects phrase correctly', () => {
    const { container } = render(
      <PhraseSelector
        value={"Interstellar"}
        phrase="int"
      />
    );

    expect(container).toMatchSnapshot();
  });

  test(`doesn't return selected text if phrase wasn't found`, () => {
    const { container } = render(
      <PhraseSelector
        value={"Interstellar"}
        phrase="abc"
      />
    );
    
    expect(container).toMatchSnapshot();
  });
});