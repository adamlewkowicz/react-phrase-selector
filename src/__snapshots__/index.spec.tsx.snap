// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhraseSelector /> doesn't return selected text if phrase wasn't found 1`] = `
<div>
  Interstellar
</div>
`;

exports[`<PhraseSelector /> selects phrase correctly 1`] = `
<div>
  
  <strong
    class="react-phrase-selector"
  >
    Int
  </strong>
  erstellar
</div>
`;
